joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -197,161 +205,162 @@ controller_interface::return_type JointTrajectoryController::update(
       (*traj_point_active_ptr_)
         ->sample(time, interpolation_method_, state_desired_, start_segment_itr, end_segment_itr);
 
-    if (valid_point)
+    if (!valid_point)
+    {
+      return controller_interface::return_type::ERROR;
+    }
+
+    bool tolerance_violated_while_moving = false;
+    bool outside_goal_tolerance = false;
+    bool within_goal_time = true;
+    double time_difference = 0.0;
+    const bool before_last_point = end_segment_itr != (*traj_point_active_ptr_)->end();
```

Comment:
Nitpick, for readability

```suggestion
    const bool before_last_point = (end_segment_itr != (*traj_point_active_ptr_)->end());
```